packages/devtools/web/devtools_analytics.js

```diff
@@ -4,145 +4,84 @@
 
 // URI parameter to control DevTools analytics collection:
 //
-//   &gtags=enabled                 // Analytics will run but ga-confirm and opt-in must work (see ga-confirm below).
-//   &gtags=disabled                // Default is disabled implies no GA will run (even opt-in dialog won't display).
-//
-//   &ga-confirm=?                  // Displays the GA collection confirmation dialog, to change answer.
-//   &ga-confirm=reset              // Resets/deletes user's collection answer in local storage.
-//   &ga-confirm=true               // Analytics COLLECT opt-in answer from an outside application e.g., VSCode.
-//   &ga-confirm=false              // Analytics DO NOT COLLECT opt-in answer from an outside application e.g., VSCode.
+//   &gtags=enabled                 // Analytics will display opt-in dialog, if accepted when enabled collects GA.
+//   &gtags=disabled                // Default is disabled implies no GA, opt-in not displayed even if opt-in accepted
+//                                  // gtags=disabled GA will NEVER be collected.
+//   &gtags=reset                   // Resets/deletes user's collection answer in local storage.
 
+// Used for GA collecting (communicating to https://www.googletagmanager.com/gtag/js script).
 window.dataLayer = window.dataLayer || [];
-
 function gtag() {
-    if (gtagsEnabled()) {
-        dataLayer.push(arguments);
-    }
-}
-
-function _getUrlVars() {
-    let allVars = {};
-    window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(_, key, value) {
-        allVars[key] = value;
-    });
-    return allVars;
+  if (gtagsEnabled()) {
+      dataLayer.push(arguments);
+  }
 }
 
-let _urlVars = _getUrlVars();
+// Parse the URI parameters.
+let _gtagsValue = new URLSearchParams(window.location.search).get('gtags');
+let _gtagsReset = _gtagsValue == 'reset';
+let _gtagsEnabled = _gtagsValue == 'enabled' ? true : false;
+// Default is disabled if &gtags= is not specified
+let _gtagsDisabled = _gtagsValue == null || _gtagsValue == 'disabled' ? true : false;
```

Comment:
the `? true : false` can be removed. We should have a lint that catches unneeded expressions like this. I think Java does. @pq

Reply:
good catch!

[`avoid_bool_literals_in_conditional_expressions`](http://dart-lang.github.io/linter/lints/avoid_bool_literals_in_conditional_expressions.html) should flag that one.

(not sure why it's not in pedantic?)

Reply:
This is JS code.  I'll remove the ? :